jss-sparsegl/script_for_plotting.R

```diff
@@ -0,0 +1,134 @@
+#                          times = 1)))
+#       }
+#       df[[k]] <- df[[k]] %>%
+#         select(expr, mean) %>%
+#         mutate(
+#           change_feature = factor(change_feature),
+#           change_list := factor(!!rlang::sym(change_feature)),
+#           within_group = within_group)
+#       k <- k + 1
+#       print(k)
+#     }
+#   }
+#   datf <- bind_rows(df)
+#   return(datf)
+# }
+
+
+plot_box <- function(df) {
+  df <- df %>%
+    group_by(expr, change_feature, change_list, within_group, type) %>% 
+    summarise(mean = mean(mean), .groups = 'drop')
+  plot <- df %>% ggplot(aes(x = change_list, y = mean, color = expr, group = expr)) +
+    geom_point() +
+    geom_line() + 
+    theme_bw() + 
+    theme(legend.position = "bottom", legend.title = element_blank(), axis.title.x = element_blank()) +
+    xlab("Change values") +
+    ylab("Running Time (ms)") +
+    facet_grid(type + within_group ~ change_feature, 
+               labeller = as_labeller(c('p' = "Number of features p", 'n' = "Sample size n",
+                                        'TRUE' = "Within-group sparsity", 'FALSE' = "Without-group sparsity",
+                                        'linear' = "Linear regression", 'logistic' = "Logistic regression")), switch = "x") +
+    scale_colour_viridis_d() +
+    scale_y_log10() 
+  plot
+}
+
+
+
+# within_p <- simulate(change_list = c(20, 60, 100, 200, 300, 500), change_feature = "p", within_group = TRUE, group_size = 5, family = "gaussian")
+# within_n <- simulate(change_list = c(20, 60, 100, 200, 300, 500), change_feature = "n", within_group = TRUE, group_size = 5, family = "gaussian")
+# without_p <- simulate(change_list = c(20, 60, 100, 200, 300, 500), change_feature = "p", within_group = FALSE, group_size = 5, family = "gaussian")
+# without_n <- simulate(change_list = c(20, 60, 100, 200, 300, 500), change_feature = "n", within_group = FALSE, group_size = 5, family = "gaussian")
+# 
+# 
+# 
+# log_within_p <- simulate(change_list = c(20, 60, 100, 200, 300, 500), change_feature = "p", within_group = TRUE, group_size = 5, family = "binomial")
+# log_within_n <- simulate(change_list = c(20, 60, 100, 200, 300, 500), change_feature = "n", within_group = TRUE, group_size = 5, family = "binomial")
+# log_without_p <- simulate(change_list = c(20, 60, 100, 200, 300, 500), change_feature = "p", within_group = FALSE, group_size = 5, family = "binomial")
+# log_without_n <- simulate(change_list = c(20, 60, 100, 200, 300, 500), change_feature = "n", within_group = FALSE, group_size = 5, family = "binomial")
+# 
+# 
+# 
+# df1 <- bind_rows(within_p, within_n, without_p, without_n) %>% mutate(type = "linear")
+# df2 <- bind_rows(log_within_p, log_within_n, log_without_p, log_without_n) %>% mutate(type = "logistic")
+# df <- bind_rows(df1, df2)
+# saveRDS(df, "/Users/xiaoxuanliang/Desktop/sparsegl/jss-sparsegl/running-time-records-current")
```

Comment:
1. Save the file with `.rds` extension.
2. Don't use your whole (local) path. Use the `{here}` package to create a relative path for reproducibility. See [the website](https://here.r-lib.org)

Reply:
The warning messages always pop up when compiling if saving it as a .rds file but no warning got generated without this extension.

Reply:
Files without extensions are hard to know what they are. What warning messages?

Reply:
popped messages: 
Attaching package: 'dplyr'
The following objects are masked from 'package:stats':
    filter, lag
The following objects are masked from 'package:base':
    intersect, setdiff, setequal, union

Reply:
Those are "messages" not "warnings". Basically, they mean that some functions in `dplyr` have the same name as those in `stats` or `base`. But since you loaded `dplyr` second, R will assume that if you call `lag()`, you wanted `dplyr::lag()` rather than `stats::lag()`.

You can prevent these from printing in the Rmd with `message = FALSE` in the chunk.